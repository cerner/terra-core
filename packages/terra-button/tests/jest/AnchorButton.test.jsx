import React from 'react';

import AnchorButton from '../../src/variants/_AnchorButton';

// Snapshot Tests
it('should render as an anchor when AnchorButton is used', () => {
  const anchorButton = shallow(<AnchorButton text="href" href="MockHref" />);
  expect(anchorButton).toMatchSnapshot();
});

// Structure
it('should render as an anchor tag when AnchorButton is used', () => {
  const anchorButton = shallow(<AnchorButton href="MockHref" text="text" />);
  expect(anchorButton.is('a')).toEqual(true);
});
