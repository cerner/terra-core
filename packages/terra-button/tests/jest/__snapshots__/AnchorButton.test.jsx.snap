// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render as an anchor when AnchorButton is used 1`] = `
<a
  aria-disabled={false}
  aria-label="href"
  className="button neutral"
  href="MockHref"
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  role="button"
  title="href"
  type="button"
>
  <span
    className="button-label text-only"
  >
    <span>
      href
    </span>
  </span>
</a>
`;
